tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc

```diff
@@ -391,12 +403,14 @@ Status TRTEngineOp::VerifyInputShapes(const std::vector<TensorShape>& shapes) {
                                    TensorShapeUtils::ShapeListString(shapes));
   }
 
-  const int batch_size = shapes[0].dim_size(0);
-  for (const TensorShape& shape : shapes) {
-    if (shape.dims() < 1 || batch_size != shape.dim_size(0)) {
-      return errors::InvalidArgument(
-          "Input shapes are inconsistent on the batch dimension, for ", name(),
-          ": ", TensorShapeUtils::ShapeListString(shapes));
+  if (use_implicit_batch_) {
+    const int batch_size = shapes[0].dim_size(0);
+    for (const TensorShape& shape : shapes) {
+      if (shape.dims() < 1 || batch_size != shape.dim_size(0)) {
```

Comment:
Can you explain the check for shape.dims() < 1? Why it is not hoisted outside the loop, which is equivalent to checking variable batch_size < 1?